### ISSUE TITLE
Bug: missing dependencies on 2.0.0a3

### ISSUE THREAD

**Issue by elip-OE** [litestar-org/litestar#1478]
### Description

version 2.0.0a3 is missing some dependencies

I had to install manually the following packages: 
1. pytest
2. click
3. rich
4. jsbeautifier
5. uvicorn

 in order to run `litestar --app app:app run`

### MCVE

```python
from typing import Dict

from litestar import Litestar, get

@get("/")
def hello_world() -> Dict[str, str]:
    """Keeping the tradition alive with hello world."""
    return {"hello": "world"}


app = Litestar(route_handlers=[hello_world])
```

### Steps to reproduce

```bash
1. use clean virtualenv
2. run `pip install litestar==2.0.0a3`
3. create the demo app from the README file
4. run `litestart --app app:app run`
```

### Logs

```bash
$ litestar --app app:app run
Traceback (most recent call last):
  File "/home/test/.venv/bin/litestar", line 5, in <module>
    from litestar import __main__
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/__init__.py", line 1, in <module>
    from litestar.app import Litestar
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/app.py", line 13, in <module>
    from litestar._openapi.path_item import create_path_item
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/_openapi/path_item.py", line 6, in <module>
    from litestar._openapi.parameters import create_parameter_for_handler
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/_openapi/parameters.py", line 5, in <module>
    from litestar._openapi.schema_generation import create_schema
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/_openapi/schema_generation/__init__.py", line 1, in <module>
    from .schema import create_schema
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/_openapi/schema_generation/schema.py", line 39, in <module>
    from litestar._openapi.schema_generation.examples import create_examples_for_field
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/_openapi/schema_generation/examples.py", line 7, in <module>
    from polyfactory.exceptions import ParameterException
  File "/home/test/.venv/lib/python3.8/site-packages/polyfactory/__init__.py", line 2, in <module>
    from .factories import BaseFactory, DataclassFactory, TypedDictFactory
  File "/home/test/.venv/lib/python3.8/site-packages/polyfactory/factories/__init__.py", line 1, in <module>
    from polyfactory.factories.base import BaseFactory, DataclassFactory, TypedDictFactory
  File "/home/test/.venv/lib/python3.8/site-packages/polyfactory/factories/base.py", line 51, in <module>
    from polyfactory.fields import Fixture, Ignore, PostGenerated, Require, Use
  File "/home/test/.venv/lib/python3.8/site-packages/polyfactory/fields.py", line 6, in <module>
    from polyfactory.pytest_plugin import FactoryFixture
  File "/home/test/.venv/lib/python3.8/site-packages/polyfactory/pytest_plugin.py", line 14, in <module>
    from pytest import fixture
ModuleNotFoundError: No module named 'pytest'



$ litestar --app app:app run
Traceback (most recent call last):
  File "/home/test/.venv/bin/litestar", line 5, in <module>
    from litestar import __main__
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/__main__.py", line 1, in <module>
    from litestar.cli.main import litestar_group  # pragma: no cover
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/main.py", line 3, in <module>
    from click import Context, group, option, pass_context
ModuleNotFoundError: No module named 'click'


$ litestar --app app:app run
Traceback (most recent call last):
  File "/home/test/.venv/bin/litestar", line 5, in <module>
    from litestar import __main__
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/__main__.py", line 1, in <module>
    from litestar.cli.main import litestar_group  # pragma: no cover
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/main.py", line 5, in <module>
    from ._utils import LitestarEnv, LitestarExtensionGroup
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/_utils.py", line 14, in <module>
    from rich.console import Console
ModuleNotFoundError: No module named 'rich'


$ litestar --app app:app run
Traceback (most recent call last):
  File "/home/test/.venv/bin/litestar", line 5, in <module>
    from litestar import __main__
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/__main__.py", line 1, in <module>
    from litestar.cli.main import litestar_group  # pragma: no cover
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/main.py", line 6, in <module>
    from .commands import core, schema, sessions
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/commands/schema.py", line 6, in <module>
    from jsbeautifier import Beautifier
ModuleNotFoundError: No module named 'jsbeautifier'

$ litestar --app app:app run
Using Litestar app from env: 'app:app'
┌──────────────────┬──────────────────────┐
│ Litestar version │ 2.0.0                │
│ Debug mode       │ Disabled             │
│ CORS             │ Disabled             │
│ CSRF             │ Disabled             │
│ OpenAPI          │ Enabled path=/schema │
│ Compression      │ Disabled             │
└──────────────────┴──────────────────────┘
Starting server process ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Traceback (most recent call last):
  File "/home/test/.venv/bin/litestar", line 5, in <module>
    from litestar import __main__
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/__main__.py", line 3, in <module>
    litestar_group()  # pragma: no cover
  File "/home/test/.venv/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/test/.venv/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/test/.venv/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/test/.venv/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/test/.venv/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/test/.venv/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/_utils.py", line 202, in wrapped
    return func(*args, **kwargs)
  File "/home/test/.venv/lib/python3.8/site-packages/litestar/cli/commands/core.py", line 98, in run_command
    subprocess.run(["uvicorn", env.app_path, *_convert_uvicorn_args(process_args)], check=True)
  File "/usr/lib/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'uvicorn'
```

### Litestar Version

2.0..0a3

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

**Comment by provinzkraut** [comment]
Did you install litestar with the required extras [as specified in the documentation](https://docs.litestar.dev/2/usage/cli.html#enabling-the-cli)?

`pip install litestar[cli] --pre`

**Comment by elip-OE** [comment]
no, I've used the command from README file (which contains a mistake in the version name):

```
pip install litestar==2.0.0a3  # in the README it is 2.0.0alpha3, but I've managed to understand this
```

after your ocmment, I've used the following:
```
pip install litestar[cli]==2.0.0a3
```

and only `pytest` and `uvicorn` are missing

**Comment by provinzkraut** [comment]
> no, I've used the command from README file 

It would get a bit unwieldy to put everything in the readme file,, which is why the CLI, and the is not mentioned there in general. But you'll find that the usage guide of our docs, the CLI mentions this in the first paragraph, as well as the first page of the documentation in the "Installation" section. 

Could you give me any pointers how this could have been documented better?

> (which contains a mistake in the version name)

This is not a mistake, but an artifact of how PyPi handles versioning. If you look at this line
https://github.com/litestar-org/litestar/blob/7c990386e677c92252558b681ff4ecae4f7c63d9/pyproject.toml#L3
you'll see that the version is actually as specified, and running `pip install litestar==2.0.0alpha3` will work just fine.


<hr>

As for the missing dependencies, I'm not sure why `pytest` is required at all. It's not required by the CLI and this shouldn't occur. I will have to look into this.

**Comment by lqmanh** [comment]
This happens to me as well, at least with pytest.

I installed `litestar[standard]`, managed via Poetry.

**Comment by provinzkraut** [comment]
@lqmanh this pytest thing has been popping up a few times now. Your assessment that it’s related to polyfactory seems correct. Do you mind opening a separate issue for this?

**Comment by lqmanh** [comment]
@provinzkraut Then I should open a separate issue here or in polyfactory repo?

**Comment by provinzkraut** [comment]
@lqmanh @elip-OE both dependency issue will be fixed in the `alpha4` release which will happen today and the `1.51.10` release, which we're also planning to release later this day.

**Comment by elip-OE** [comment]
@provinzkraut many thanks for your quick response

**Comment by provinzkraut** [comment]
This has been fixed in the latest pre-release `v2.0.0alpha4`.
